<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <comment>
  Copyright (c) the JPEG XL Project Authors. All rights reserved.

  Use of this source code is governed by a BSD-style
  license that can be found in the LICENSE file.
 </comment>
 <class>FlickerTestWizard</class>
 <widget class="QWizard" name="FlickerTestWizard">
  <property name="windowTitle">
   <string>New flicker test</string>
  </property>
  <property name="options">
   <set>QWizard::NoBackButtonOnStartPage</set>
  </property>
  <widget class="QWizardPage" name="pathsPage">
   <layout class="QFormLayout" name="formLayout">
    <item row="0" column="0">
     <widget class="QLabel" name="originalFolderPromptLabel">
      <property name="text">
       <string>Folder with the original images:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
      <item>
       <widget class="QLineEdit" name="originalFolder"/>
      </item>
      <item>
       <widget class="QToolButton" name="originalFolderBrowseButton">
        <property name="text">
         <string>Browse…</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="alteredFolderPromptLabel">
      <property name="text">
       <string>Folder with the altered images:</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,0">
      <item>
       <widget class="QLineEdit" name="alteredFolder"/>
      </item>
      <item>
       <widget class="QToolButton" name="alteredFolderBrowseButton">
        <property name="text">
         <string>Browse…</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="outputFilePromptLabel">
      <property name="text">
       <string>CSV file in which to save the results:</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,0">
      <item>
       <widget class="QLineEdit" name="outputFile"/>
      </item>
      <item>
       <widget class="QToolButton" name="outputFileBrowseButton">
        <property name="text">
         <string>Browse…</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="timesPage">
   <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,0,1">
    <item>
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </spacer>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,0,1">
      <item>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="advanceTimePromptLabel">
          <property name="text">
           <string>Advance time:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="advanceTime">
          <property name="suffix">
           <string> ms</string>
          </property>
          <property name="minimum">
           <number>100</number>
          </property>
          <property name="maximum">
           <number>3000</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="viewingTimePromptLabel">
          <property name="text">
           <string>Viewing time (t&lt;sub&gt;VIEW&lt;/sub&gt;):</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="viewingTime">
          <property name="specialValueText">
           <string>no limit</string>
          </property>
          <property name="suffix">
           <string> s</string>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>30</number>
          </property>
          <property name="value">
           <number>4</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="blankingTimePromptLabel">
          <property name="text">
           <string>Blanking time (t&lt;sub&gt;BLANK&lt;/sub&gt;):</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="blankingTime">
          <property name="suffix">
           <string> ms</string>
          </property>
          <property name="minimum">
           <number>50</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="singleStep">
           <number>50</number>
          </property>
          <property name="value">
           <number>250</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="grayFlickering">
        <property name="title">
         <string>Gray flickering</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QFormLayout" name="formLayout_4">
           <item row="0" column="0">
            <widget class="QLabel" name="grayFadingTimePromptLabel">
             <property name="text">
              <string>Fading time to and from gray:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="grayFadingTime">
             <property name="suffix">
              <string> ms</string>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
             <property name="singleStep">
              <number>100</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="grayTimePromptLabel">
             <property name="text">
              <string>Time on gray:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="grayTime">
             <property name="suffix">
              <string> ms</string>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
             <property name="singleStep">
              <number>100</number>
             </property>
             <property name="value">
              <number>300</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="totalGrayTimeLabel">
           <property name="text">
            <string>Total gray time: 500 ms</string>
           </property>
           <property name="textFormat">
            <enum>Qt::RichText</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QDialogButtonBox" name="timingButtonBox">
        <property name="standardButtons">
         <set>QDialogButtonBox::RestoreDefaults</set>
        </property>
        <property name="centerButtons">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="intensityTargetPage">
   <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="1,0,1">
    <item>
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </spacer>
    </item>
    <item>
     <layout class="QFormLayout" name="formLayout_5">
      <item row="0" column="0">
       <widget class="QLabel" name="intensityTargetPromptLabel">
        <property name="text">
         <string>Display peak luminance:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="intensityTarget">
        <property name="correctionMode">
         <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
        </property>
        <property name="suffix">
         <string> cd/m²</string>
        </property>
        <property name="minimum">
         <number>20</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="stepType">
         <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
        </property>
        <property name="value">
         <number>250</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="horizontalSpacer_4">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="spacingPage">
   <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,0,0">
    <item>
     <widget class="jxl::SplitView" name="spacingDemo" native="true"/>
    </item>
    <item>
     <spacer name="verticalSpacer_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </spacer>
    </item>
    <item>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="spacingPromptLabel">
        <property name="text">
         <string>Spacing between the images:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,0">
        <item>
         <widget class="QSlider" name="spacing">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spacingSpinBox">
          <property name="suffix">
           <string> px</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>jxl::SplitView</class>
   <extends>QWidget</extends>
   <header>tools/flicker_test/split_view.h</header>
   <container>1</container>
   <slots>
    <slot>setSpacing(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>spacing</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spacingDemo</receiver>
   <slot>setSpacing(int)</slot>
  </connection>
  <connection>
   <sender>spacing</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spacingSpinBox</receiver>
   <slot>setValue(int)</slot>
  </connection>
  <connection>
   <sender>spacingSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spacing</receiver>
   <slot>setValue(int)</slot>
  </connection>
 </connections>
</ui>
